main.go

```diff
@@ -77,6 +95,28 @@ func main() {
 		klog.FlushAndExit(klog.ExitFlushTimeout, 1)
 	}
 
+	if enableLeaderElection {
+		logger.Info("Leader election is enabled.")
+		go leaderelection.StartLeaderElection(ctx, kubeClient, cfg, logger, cancel, runControllers, leaderelection.Config{
+			LeaseDuration: leaseDuration,
+			RenewDeadline: renewDeadline,
+			RetryPeriod:   retryPeriod,
+			ResourceLock:  resourceLock,
+			ResourceName:  resourceName,
+		})
+	} else {
+		logger.Info("Leader election is disabled.")
+		go runControllers(ctx, kubeClient, cfg, logger)
+	}
+
+	<-ctx.Done()
+	logger.Info("Shutting down server")
```

Comment:
```suggestion
```

you don't need to wait `<-ctx.Done()` before server shutdown, because the server does it.

Reply:
removed that part